src/test/java/staffconnect/logic/parser/AddMeetingCommandParserTest.java

```diff
@@ -0,0 +1,107 @@
+package staffconnect.logic.parser;
+
+
+import static staffconnect.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static staffconnect.logic.commands.CommandTestUtil.DESC_MIDTERM;
+import static staffconnect.logic.commands.CommandTestUtil.INVALID_DESCRIPTION;
+import static staffconnect.logic.commands.CommandTestUtil.INVALID_STARTDATE;
+import static staffconnect.logic.commands.CommandTestUtil.MEETDATE_STARTDATE;
+import static staffconnect.logic.commands.CommandTestUtil.VALID_MEETING;
+import static staffconnect.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static staffconnect.logic.parser.CliSyntax.PREFIX_STARTDATE;
+import static staffconnect.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static staffconnect.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static staffconnect.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static staffconnect.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+
+import org.junit.jupiter.api.Test;
+
+import staffconnect.commons.core.index.Index;
+import staffconnect.logic.Messages;
+import staffconnect.logic.commands.AddMeetingCommand;
+import staffconnect.model.meeting.Description;
+import staffconnect.model.meeting.MeetDateTime;
+
+public class AddMeetingCommandParserTest {
+
+    private static final String MEETING_MISSING_INDEX = " meeting at finals";
+
+    private static final String MESSAGE_INVALID_FORMAT =
+        String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddMeetingCommand.MESSAGE_USAGE);
+
+    private final AddMeetingCommandParser parser = new AddMeetingCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, MEETING_MISSING_INDEX, MESSAGE_INVALID_FORMAT);
+
+        // no index and no field specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + DESC_MIDTERM + MEETDATE_STARTDATE, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + DESC_MIDTERM + MEETDATE_STARTDATE, MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidValue_failure() {
+
+        // invalid description followed by valid date
+        assertParseFailure(parser, "1" + INVALID_DESCRIPTION + MEETDATE_STARTDATE,
+                           Description.MESSAGE_CONSTRAINTS);
+        // valid description followed by invalid date
+        assertParseFailure(parser, "1" + DESC_MIDTERM + INVALID_STARTDATE, MeetDateTime.MESSAGE_CONSTRAINTS);
+    }
```

Comment:
Perhaps you could add in a test case for `two invalid values, only first invalid value reported` and `non-empty preamble` here

Reply:
Added ```two invalid values``` and ```non-empty preamble```, I don't think ```first invalid value reported ``` is applicable in my case. Because it will fall into its children class error, or and invalid command if either one of the values are invalid

Reply:
Fixed